presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -991,6 +992,15 @@ protected Scope visitExplain(Explain node, Optional<Scope> scope)
         @Override
         protected Scope visitQuery(Query node, Optional<Scope> scope)
         {
+            Statement statement = analysis.getStatement();
```

Comment:
Since we are using statement from the original analysis object, this would never truly switch to MV query, as it would point to base query.